src/stan/math/prim/mat/meta/get.hpp

```diff
@@ -2,11 +2,12 @@
 #define STAN__MATH__PRIM__MAT__META__GET_HPP
 
 #include <stan/math/prim/mat/fun/Eigen.hpp>
+#include <stan/math/prim/mat/meta/value_type.hpp>
 
 namespace stan {
 
-  template <typename T, int R, int C>
-  inline T get(const Eigen::Matrix<T,R,C>& m, size_t n) {
+  template <typename T>
+  inline typename stan::math::value_type<T>::type get(const Eigen::MatrixBase<T>& m, size_t n) {
```

Comment:
is the call through the metaprogram `value_type` necessary here? What type is `T` here? (these are actual questions -- I don't know the answer to this)